File: frontend/src/services/departmentService.js
import api from './api'

export const departmentService = {
  getDepartments: async () => {
    const response = await api.get('/api/departments')
    return response.data
  },

  getDepartmentById: async (id) => {
    const response = await api.get(`/api/departments/${id}`)
    return response.data
  },

  createDepartment: async (deptData) => {
    const response = await api.post('/api/departments', deptData)
    return response.data
  },

  updateDepartment: async (id, deptData) => {
    const response = await api.put(`/api/departments/${id}`, deptData)
    return response.data
  },

  deleteDepartment: async (id) => {
    const response = await api.delete(`/api/departments/${id}`)
    return response.data
  },

  getDepartmentEmployees: async (id) => {
    const response = await api.get(`/api/departments/${id}/employees`)
    return response.data
  }
}

export default departmentService

File: backend/src/main/java/com/erp/service/DepartmentService.java
package com.erp.service;

import com.erp.dto.request.DepartmentRequestDto;
import com.erp.dto.response.DepartmentResponseDto;
import com.erp.dto.response.EmployeeResponseDto;

import java.util.List;

public interface DepartmentService {
    List<DepartmentResponseDto> getAllDepartments();
    DepartmentResponseDto getDepartmentById(Long id);
    DepartmentResponseDto createDepartment(DepartmentRequestDto dto);
    DepartmentResponseDto updateDepartment(Long id, DepartmentRequestDto dto);
    void deleteDepartment(Long id);
    List<EmployeeResponseDto> getDepartmentEmployees(Long id);
}
